src/Microsoft.DotNet.Build.Tasks/PackageFiles/packageresolve.targets

```diff
@@ -124,18 +124,36 @@
                                        ValidationPatterns="@(ValidationPattern)" />
   </Target>
 
-  <Target Name="FilterTargetingPackResolvedNugetPackages" AfterTargets="ResolveNuGetPackages"
-          Condition="'$(TargetingPackNugetPackageId)' != ''">
+  <Target Name="FilterTargetingPackResolvedNugetPackages" AfterTargets="ResolveNuGetPackages" >
+    <PropertyGroup>
+      <_TargetingPackPrefix>Microsoft.TargetingPack</_TargetingPackPrefix>
+    </PropertyGroup>
+
+    <!--
+      Add the core libraries and windows to the reference set by default to avoid a lot of duplication in projects
+      They only act as a filter so if they aren't present in the packages references it will not impact anything.
+    -->
+    <ItemGroup Condition="'$(ExcludeDefaultTargetingPackReferences)' != 'true'">
+      <TargetingPackReference Include="mscorlib" />
+      <TargetingPackReference Include="System.Private.CoreLib" />
```

Comment:
Good point. That will cause issues for System.Private.CoreLib because of the mix of contracts and not in that targeting pack. I will remove this one. The biggest reason I added these defaults were for mscorlib and windows as with my change to always filter it will break projects because there are a lot of them that don't have a TargetingPackReference for those as it wasn't required before because we weren't filtering those 2 packages in most cases.